### PR TITLE
Create cache-handlers.md

### PR DESCRIPTION
**Pull request by Ben-MyBB**
This is still a work in progress.

### CODE REVIEW

**Comment by Ben-MyBB** [comment]
@euantorano can you have a quick read through when you get a sec? If this is the sort of thing we want to go for I can finish this off (still have things to add like how to select each one).

**Comment by euantorano** [comment]
Yep, will have a look today thanks.

File: 1.8/administration/cache-handlers.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+The Alternative PHP Cache (APC) is a free and open opcode cache for PHP. Its goal is to provide a free, open, and robust framework for caching and optimizing PHP intermediate code. 
+
+https://www.php.net/manual/en/book.apc.php
+
+Please note, APC Cache Handler is no longer officially supported by PHP.
+
+**Disk Cache Handler**
+
+RFC 2616 compliant caching provides a mechanism to verify whether stale or expired content is still fresh, and can represent a significant performance boost when the origin server supports conditional requests by honouring the If-None-Match HTTP request header. Content is only regenerated from scratch when the content has changed, and not when the cached entry expires.
```
Comment (euantorano) [inline 1.8/administration/cache-handlers.md]: Our cache handler is just a file based cache. I'd suggest something like:

```suggestion
Cache file are written to the filesystem within the `cache/` directory as PHP files. The `cache/` directory must be writeable for this cache handler to function.
```

File: 1.8/administration/cache-handlers.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+https://httpd.apache.org/docs/2.4/mod/mod_cache.html
+
+**eAccelerator**
+
+eAccelerator is a free open-source PHP accelerator & optimizer. It increases the performance of PHP scripts by caching them in their compiled state, so that the overhead of compiling is almost completely eliminated. It also optimizes scripts to speed up their execution. eAccelerator typically reduces server load and increases the speed of your PHP code by 1-10 times. 
+
+https://github.com/eaccelerator/eaccelerator/wiki
+
+Please note, eAccelerator is no longer officially supported.
```
Comment (euantorano) [inline 1.8/administration/cache-handlers.md:36]: Supported by MyBB, not by upstream:
```suggestion
Please note, eAccelerator is no longer officially supported by the upstream developers.
```

**Review by euantorano — CHANGES_REQUESTED** [review]
Looks good so far, just some minor comments.

File: 1.8/administration/cache-handlers.md [hunk 3]
```diff
@@ -12,11 +12,11 @@ The Alternative PHP Cache (APC) is a free and open opcode cache for PHP. Its goa
 
 https://www.php.net/manual/en/book.apc.php
 
-Please note, APC Cache Handler is no longer officially supported by PHP.
+Please note, APC Cache Handler is no longer officially supported.
 
 **Disk Cache Handler**
 
-RFC 2616 compliant caching provides a mechanism to verify whether stale or expired content is still fresh, and can represent a significant performance boost when the origin server supports conditional requests by honouring the If-None-Match HTTP request header. Content is only regenerated from scratch when the content has changed, and not when the cached entry expires.
+Cache file are written to the filesystem within the `cache/` directory as PHP files. The `cache/` directory must be writeable for this cache handler to function.
```
Comment (euantorano) [inline 1.8/administration/cache-handlers.md]:
```suggestion
Cache files are written to the filesystem within the `cache/` directory as PHP files. The `cache/` directory must be writeable for this cache handler to function.
```

**Review by euantorano — CHANGES_REQUESTED** [review]
Sorry, minor spelling/grammar tweak.

File: 1.8/administration/cache-handlers.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+$config['cache_store'] = 'db';
+</table>
+
+You then need to edit the value on line 70. For example, if you wished to use Redis you change line 70 too:
+
+<table>
+  $config['cache_store'] = 'redis';
+</table>
+
+Don't forget, if you wish to use Redis or Memcache you will also need to set the Hostname and Port settings which are also set in the config.php file. The setting for Memcache can be found at line 50 and the settings for Redis can be found at line 62. You will edit the Hostname and Port number values similiar to above.
```
Comment (euantorano) [inline 1.8/administration/cache-handlers.md]:
```suggestion
Don't forget, if you wish to use Redis or Memcache you will also need to set the Hostname and Port settings which are also set in the config.php file. The settings for Memcache can be found at line 50 and the settings for Redis can be found at line 62. You will edit the Hostname and Port number values similiar to above.
```

**Review by euantorano — CHANGES_REQUESTED** [review]

**Comment by euantorano** [comment]
This looks good, just a couple of minor tweaks.

**Comment by dvz** [comment]
Are `mod_cache`, https://redis.io/topics/client-side-caching relevant here?

**Comment by euantorano** [comment]
I don't think so - this is specifically for the built in cache interface for back-end caching.

**Comment by Ben-MyBB** [comment]
Thanks @euantorano. Applied both of your suggestions.

**Comment by euantorano** [comment]
Awesome, thanks. I’ll have another look over, hen this should be good to merge I believe?

File: 1.8/administration/cache-handlers.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+https://www.php.net/manual/en/book.apcu.php
+
+Please note, APCu is available in MyBB versions 1.8.23 onwards. 
+
+**Disk Cache Handler**
+
+Cache files are written to the filesystem within the `cache/` directory as PHP files. The `cache/` directory must be writeable for this cache handler to function.
+
+https://httpd.apache.org/docs/2.4/mod/mod_cache.html
```
Comment (euantorano) [inline 1.8/administration/cache-handlers.md]: Probably best to remove this link, as `mod_cache` is something different. 

```suggestion
```

File: 1.8/administration/cache-handlers.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+https://www.php.net/manual/en/book.memcache.php
+
+**Redis**
+
+Redis has a great article that explains in full the ins and outs of the Cache Handler - https://redis.io/topics/client-side-caching .
+
+https://redis.io/documentation
+
+Please note, Redis will only work with installs running MyBB version 1.8.23 or higher. You will also need to set the Redis hostname and port setting for this Cache Handler to work.
+
```
Comment (euantorano) [inline 1.8/administration/cache-handlers.md]: Might be best to add a horizontal rule here.

```suggestion
—————
```

**Comment by Ben-MyBB** [comment]
Yep, I don't think I have anything else to add.

**Review by euantorano — CHANGES_REQUESTED** [review]

**Comment by Ben-MyBB** [comment]
@euantorano updated :)

**Pull request closed without merging**